### PR TITLE
Fix Fuse user group and add Fuse open integration tests

### LINKED ISSUE #14976: Fuse failed to set user group
**Alluxio Version:**
Alluxio 2.7.3

**Describe the bug**
Caused by: alluxio.exception.AlluxioException: Could not update owner:group for /openWriteExistingWithTruncFlag to root:1001. alluxio.exception.status.FailedPreconditionException: Owner root does not belong to the group 1001
at alluxio.exception.status.AlluxioStatusException.toAlluxioException(AlluxioStatusException.java:105)
at alluxio.client.file.BaseFileSystem.rpc(BaseFileSystem.java:588)
at alluxio.client.file.BaseFileSystem.setAttribute(BaseFileSystem.java:452)
at alluxio.fuse.auth.SystemUserGroupAuthPolicy.setUserGroupIfNeeded(SystemUserGroupAuthPolicy.java:106)
at alluxio.fuse.AlluxioJniFuseFileSystem.createInternal(AlluxioJniFuseFileSystem.java:234)

Caused by: java.io.IOException: uid is 0, gid is 0, user name is root, group name is 1001, DEFAULT USER NAME is 1001, DEFAULT GROUP NAME is 1001, default user id 1001, default group id 0

Alluxio Fuse use system user name as group name, which is wrong
and thus get the wrong default group id.

**To Reproduce**
Steps to reproduce the behavior (as minimally and precisely as possible)

**Expected behavior**

**Urgency**
Describe the impact and urgency of the bug.

**Are you planning to fix it**
Please indicate if you are already working on a PR. 

**Additional context**

### PR DESCRIPTION
**Pull request by LuQQiu**
Fixes #14976

### What changes are proposed in this pull request?

Add integration test cases for different open flags behaviors.
Fix Fuse default user group information

### Why are the changes needed?

To prevent future open flags bugs
To reduce user doesn't belong to group problem

### Does this PR introduce any user facing changes?

No

### CODE REVIEW

**Comment by LuQQiu** [comment]
Error: 6.092 [ERROR] Failures: 
Error: 6.092 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.createWriteOpenRead
Error: 6.092 [ERROR]   Run 1: JNIFuseIntegrationTest.createWriteOpenRead:73->createTestFile:272 expected:<0> but was:<-5>
Error: 6.093 [ERROR]   Run 2: JNIFuseIntegrationTest.createWriteOpenRead:73->createTestFile:272 expected:<0> but was:<-5>
20:09:26.093 [INFO] 
Error: 6.093 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openReadWriteExisting
Error: 6.093 [ERROR]   Run 1: JNIFuseIntegrationTest.openReadWriteExisting:196->createTestFile:272 expected:<0> but was:<-5>
Error: 6.093 [ERROR]   Run 2: JNIFuseIntegrationTest.openReadWriteExisting:196->createTestFile:272 expected:<0> but was:<-5>
20:09:26.094 [INFO] 
Error: 6.094 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openReadWriteNonExisting
Error: 6.094 [ERROR]   Run 1: JNIFuseIntegrationTest.openReadWriteNonExisting:175 expected:<0> but was:<-5>
Error: 6.095 [ERROR]   Run 2: JNIFuseIntegrationTest.openReadWriteNonExisting:175 expected:<0> but was:<-5>
20:09:26.095 [INFO] 
Error: 6.095 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openReadWriteTruncExisting
Error: 6.096 [ERROR]   Run 1: JNIFuseIntegrationTest.openReadWriteTruncExisting:220->createTestFile:272 expected:<0> but was:<-5>
Error: 6.096 [ERROR]   Run 2: JNIFuseIntegrationTest.openReadWriteTruncExisting:220->createTestFile:272 expected:<0> but was:<-5>
20:09:26.097 [INFO] 
Error: 6.097 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openReadWriteTruncateZeroExisting
Error: 6.097 [ERROR]   Run 1: JNIFuseIntegrationTest.openReadWriteTruncateZeroExisting:245->createTestFile:272 expected:<0> but was:<-5>
Error: 6.098 [ERROR]   Run 2: JNIFuseIntegrationTest.openReadWriteTruncateZeroExisting:245->createTestFile:272 expected:<0> but was:<-5>
20:09:26.098 [INFO] 
Error: 6.098 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openWriteExistingWithTruncFlag
Error: 6.098 [ERROR]   Run 1: JNIFuseIntegrationTest.openWriteExistingWithTruncFlag:128->createTestFile:272 expected:<0> but was:<-5>
Error: 6.098 [ERROR]   Run 2: JNIFuseIntegrationTest.openWriteExistingWithTruncFlag:128->createTestFile:272 expected:<0> but was:<-5>
20:09:26.098 [INFO] 
Error: 6.098 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openWriteExistingWithTruncateZero
Error: 6.098 [ERROR]   Run 1: JNIFuseIntegrationTest.openWriteExistingWithTruncateZero:151->createTestFile:272 expected:<0> but was:<-5>
Error: 6.099 [ERROR]   Run 2: JNIFuseIntegrationTest.openWriteExistingWithTruncateZero:151->createTestFile:272 expected:<0> but was:<-5>
20:09:26.099 [INFO] 
Error: 6.099 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openWriteExistingWithoutTruncFlag
Error: 6.099 [ERROR]   Run 1: JNIFuseIntegrationTest.openWriteExistingWithoutTruncFlag:106->createTestFile:272 expected:<0> but was:<-5>
Error: 6.099 [ERROR]   Run 2: JNIFuseIntegrationTest.openWriteExistingWithoutTruncFlag:106->createTestFile:272 expected:<0> but was:<-5>
20:09:26.099 [INFO] 
Error: 6.100 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openWriteNonExisting
Error: 6.100 [ERROR]   Run 1: JNIFuseIntegrationTest.openWriteNonExisting:87 expected:<0> but was:<-5>
Error: 6.100 [ERROR]   Run 2: JNIFuseIntegrationTest.openWriteNonExisting:87 expected:<0> but was:<-5>

**Comment by LuQQiu** [comment]
Error: 0.426 [ERROR] Failures: 
Error: 0.426 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.createWriteOpenRead
Error: 0.427 [ERROR]   Run 1: JNIFuseIntegrationTest.createWriteOpenRead:73->createTestFile:272 expected:<0> but was:<-5>
Error: 0.434 [ERROR]   Run 2: JNIFuseIntegrationTest.createWriteOpenRead:73->createTestFile:272 expected:<0> but was:<-5>
20:09:40.434 [INFO] 
Error: 0.434 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openReadWriteExisting
Error: 0.435 [ERROR]   Run 1: JNIFuseIntegrationTest.openReadWriteExisting:196->createTestFile:272 expected:<0> but was:<-5>
Error: 0.435 [ERROR]   Run 2: JNIFuseIntegrationTest.openReadWriteExisting:196->createTestFile:272 expected:<0> but was:<-5>
20:09:40.435 [INFO] 
Error: 0.435 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openReadWriteNonExisting
Error: 0.435 [ERROR]   Run 1: JNIFuseIntegrationTest.openReadWriteNonExisting:175 expected:<0> but was:<-5>
Error: 0.436 [ERROR]   Run 2: JNIFuseIntegrationTest.openReadWriteNonExisting:175 expected:<0> but was:<-5>
20:09:40.437 [INFO] 
Error: 0.439 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openReadWriteTruncExisting
Error: 0.440 [ERROR]   Run 1: JNIFuseIntegrationTest.openReadWriteTruncExisting:220->createTestFile:272 expected:<0> but was:<-5>
Error: 0.441 [ERROR]   Run 2: JNIFuseIntegrationTest.openReadWriteTruncExisting:220->createTestFile:272 expected:<0> but was:<-5>
20:09:40.441 [INFO] 
Error: 0.441 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openReadWriteTruncateZeroExisting
Error: 0.441 [ERROR]   Run 1: JNIFuseIntegrationTest.openReadWriteTruncateZeroExisting:245->createTestFile:272 expected:<0> but was:<-5>
Error: 0.442 [ERROR]   Run 2: JNIFuseIntegrationTest.openReadWriteTruncateZeroExisting:245->createTestFile:272 expected:<0> but was:<-5>
20:09:40.442 [INFO] 
Error: 0.442 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openWriteExistingWithTruncFlag
Error: 0.442 [ERROR]   Run 1: JNIFuseIntegrationTest.openWriteExistingWithTruncFlag:128->createTestFile:272 expected:<0> but was:<-5>
Error: 0.442 [ERROR]   Run 2: JNIFuseIntegrationTest.openWriteExistingWithTruncFlag:128->createTestFile:272 expected:<0> but was:<-5>
20:09:40.443 [INFO] 
Error: 0.443 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openWriteExistingWithTruncateZero
Error: 0.443 [ERROR]   Run 1: JNIFuseIntegrationTest.openWriteExistingWithTruncateZero:151->createTestFile:272 expected:<0> but was:<-5>
Error: 0.443 [ERROR]   Run 2: JNIFuseIntegrationTest.openWriteExistingWithTruncateZero:151->createTestFile:272 expected:<0> but was:<-5>
20:09:40.443 [INFO] 
Error: 0.443 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openWriteExistingWithoutTruncFlag
Error: 0.444 [ERROR]   Run 1: JNIFuseIntegrationTest.openWriteExistingWithoutTruncFlag:106->createTestFile:272 expected:<0> but was:<-5>
Error: 0.444 [ERROR]   Run 2: JNIFuseIntegrationTest.openWriteExistingWithoutTruncFlag:106->createTestFile:272 expected:<0> but was:<-5>
20:09:40.444 [INFO] 
Error: 0.444 [ERROR] alluxio.client.fuse.JNIFuseIntegrationTest.openWriteNonExisting
Error: 0.444 [ERROR]   Run 1: JNIFuseIntegrationTest.openWriteNonExisting:87 expected:<0> but was:<-5>
Error: 0.444 [ERROR]   Run 2: JNIFuseIntegrationTest.openWriteNonExisting:87 expected:<0> but was:<-5>

**Comment by LuQQiu** [comment]
https://github.com/Alluxio/alluxio/runs/5088708606?check_suite_focus=true
Caused by: alluxio.exception.AlluxioException: Could not update owner:group for /openWriteExistingWithTruncFlag to root:1001. alluxio.exception.status.FailedPreconditionException: Owner root does not belong to the group 1001
	at alluxio.exception.status.AlluxioStatusException.toAlluxioException(AlluxioStatusException.java:105)
	at alluxio.client.file.BaseFileSystem.rpc(BaseFileSystem.java:588)
	at alluxio.client.file.BaseFileSystem.setAttribute(BaseFileSystem.java:452)
	at alluxio.fuse.auth.SystemUserGroupAuthPolicy.setUserGroupIfNeeded(SystemUserGroupAuthPolicy.java:106)
	at alluxio.fuse.AlluxioJniFuseFileSystem.createInternal(AlluxioJniFuseFileSystem.java:234)

**Comment by LuQQiu** [comment]
Caused by: java.io.IOException: uid is 0, gid is 0, user name is root, group name is 1001, DEFAULT USER NAME is 1001, DEFAULT GROUP NAME is 1001, default user id 1001, default group id 0

**Comment by LuQQiu** [comment]
https://github.com/Alluxio/alluxio/runs/5115563818?check_suite_focus=true
Error: 3.129 [ERROR] Failures: 
Error: 3.130 [ERROR] alluxio.master.AlluxioMasterProcessTest.stopAfterStandbyTransition[0]
Error: 3.130 [ERROR]   Run 1: AlluxioMasterProcessTest.stopAfterStandbyTransition:209
Error: 3.132 [ERROR]   Run 2: AlluxioMasterProcessTest.stopAfterStandbyTransition:209
20:47:03.135 [INFO] 
Error: 3.135 [ERROR] alluxio.master.AlluxioMasterProcessTest.stopAfterStandbyTransition[3]
Error: 3.135 [ERROR]   Run 1: AlluxioMasterProcessTest.stopAfterStandbyTransition:209
Error: 3.135 [ERROR]   Run 2: AlluxioMasterProcessTest.stopAfterStandbyTransition:209

**Comment by LuQQiu** [comment]
https://github.com/Alluxio/alluxio/runs/5115563818?check_suite_focus=true
Error: 3.135 [ERROR] alluxio.master.AlluxioMasterProcessTest.stopAfterStandbyTransition[3]
Error: 3.135 [ERROR]   Run 1: AlluxioMasterProcessTest.stopAfterStandbyTransition:209
Error: 3.135 [ERROR]   Run 2: AlluxioMasterProcessTest.stopAfterStandbyTransition:209
20:47:03.135 [INFO]

File: integration/fuse/src/main/java/alluxio/fuse/auth/SystemUserGroupAuthPolicy.java [hunk 1]
```diff
@@ -26,18 +26,12 @@
...
-import java.io.IOException;
-
 /**
  * Default Fuse Auth Policy.
  */
 public final class SystemUserGroupAuthPolicy implements AuthPolicy {
   private static final Logger LOG = LoggerFactory.getLogger(
       SystemUserGroupAuthPolicy.class);
-  private static final String DEFAULT_USER_NAME = System.getProperty("user.name");
-  private static final String DEFAULT_GROUP_NAME = System.getProperty("user.name");
```
Comment (LuQQiu) [inline integration/fuse/src/main/java/alluxio/fuse/auth/SystemUserGroupAuthPolicy.java:38]: The original group name is wrong

**Review by yyongycy — APPROVED** [review]
LGTM

**Comment by LuQQiu** [comment]
Error: 2.483 [ERROR] alluxio.client.cli.fs.command.CheckConsistencyCommandIntegrationTest  Time elapsed: 62.881 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: Timed out waiting for process Alluxio master @588d6c251d83:36777 to start
https://github.com/Alluxio/alluxio/runs/5150851535?check_suite_focus=true

**Comment by LuQQiu** [comment]
Error: 7.967 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.8.1:compile (default-compile) on project alluxio-table-base: Compilation failure: Compilation failure: 
Error: 7.968 [ERROR] /usr/src/alluxio/table/base/src/main/java/alluxio/table/ProtoUtils.java:[14,26] package alluxio.grpc.table does not exist
Error: 7.968 [ERROR] /usr/src/alluxio/table/base/src/main/java/alluxio/table/ProtoUtils.java:[15,26] package alluxio.grpc.table does not exist
Error: 7.968 [ERROR] /usr/src/alluxio/table/base/src/main/java/alluxio/table/ProtoUtils.java:[16,26] package alluxio.grpc.table does not exist
https://github.com/Alluxio/alluxio/runs/5150851575?check_suite_focus=true

**Comment by LuQQiu** [comment]
Error: 2.118 [ERROR] Errors: 
Error: 2.118 [ERROR] alluxio.master.AlluxioMasterProcessTest.stopAfterStandbyTransition[1]
Error: 2.118 [ERROR]   Run 1: AlluxioMasterProcessTest.stopAfterStandbyTransition:206 » Timeout Timed out wa...
Error: 2.118 [ERROR]   Run 2: AlluxioMasterProcessTest.stopAfterStandbyTransition:206 » Timeout Timed out wa...
https://github.com/Alluxio/alluxio/runs/5150851515?check_suite_focus=true

**Comment by LuQQiu** [comment]
JNIFuseIntegrationTest failed in Mac
with exception
```
	at alluxio.fuse.AlluxioJniFuseFileSystem.create(AlluxioJniFuseFileSystem.java:207)
	at alluxio.client.fuse.JNIFuseIntegrationTest.createTestFile(JNIFuseIntegrationTest.java:281)
	at alluxio.client.fuse.JNIFuseIntegrationTest.openWriteExistingWithTruncFlag(JNIFuseIntegrationTest.java:137)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
	at com.intellij.junit4.JUnit4IdeaTestRunner.startRunnerWithArgs(JUnit4IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:235)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:54)
Caused by: alluxio.exception.AlluxioException: Could not update owner:group for /openWriteExistingWithTruncFlag to alluxio:wheel. alluxio.exception.status.FailedPreconditionException: Owner alluxio does not belong to the group wheel
	at alluxio.exception.status.AlluxioStatusException.toAlluxioException(AlluxioStatusException.java:105)
	at alluxio.client.file.BaseFileSystem.rpc(BaseFileSystem.java:581)
	at alluxio.client.file.BaseFileSystem.setAttribute(BaseFileSystem.java:445)
	at alluxio.fuse.auth.SystemUserGroupAuthPolicy.setUserGroupIfNeeded(SystemUserGroupAuthPolicy.java:103)
	at alluxio.fuse.AlluxioJniFuseFileSystem.createInternal(AlluxioJniFuseFileSystem.java:226)
	... 34 more
```
Alluxio:wheel is valid user group pair in my own mac, but not quite sure why it's not valid in Alluxio system

**Comment by LuQQiu** [comment]
uid is 0, gid is 0, user name is alluxio, group name is wheel, default uid is 501, default gid is 20, default group name is staff, default user name is alluxio

**Comment by LuQQiu** [comment]
during test, fuse_get_context() may return 0 filled contents

**Pull request merged**